transport/tchannel/transport.go

```diff
@@ -333,22 +338,44 @@ func (t *Transport) onPeerStatusChanged(tp *tchannel.Peer) {
 // CreateTLSOutboundChannel creates a outbound channel for managing tls
 // connections with the given tls config and destination name.
 // Usage:
-// 	tr, _ := tchannel.NewTransport(...)
-//  outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
-//  outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
+//
+//		tr, _ := tchannel.NewTransport(...)
+//	 outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
+//	 outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
 func (t *Transport) CreateTLSOutboundChannel(tlsConfig *tls.Config, destinationName string) (peer.Transport, error) {
+	dialerFunc := t.dialer
+	if t.enableMPTCP {
```

Comment:
If t.dialer is passed in by options, those lines of code will overwrite user-passed in dialer into MPTCP dialer.
Maybe we should just change dialer when [buildTransport](https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/config.go#L156)? [buildTransport](https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/config.go#L156) will change t.dialer  directly, so we don't need to do it in buildUnaryOutbound.